src/GitHub.Exports/Extensions/SimpleRepositoryModelExtensions.cs

```diff
@@ -30,5 +35,58 @@ public static bool MightContainSolution(this ISimpleRepositoryModel repository)
                       .Any(x => ((x.Attributes.HasFlag(FileAttributes.Directory) || x.Attributes.HasFlag(FileAttributes.Normal)) &&
                                 !x.Name.StartsWith(".", StringComparison.Ordinal) && !x.Name.StartsWith("readme", StringComparison.OrdinalIgnoreCase)));
         }
+
+        public static string CurrentSha(this ISimpleRepositoryModel repository)
+        {
+            var repo = GitService.GitServiceHelper.GetRepo(repository.LocalPath);
+            return repo.Commits.FirstOrDefault()?.Sha;
+        }
+
+        public static Uri GenerateUrl(this ISimpleRepositoryModel repository, string path = null,
+            int startLine = -1, int endLine = -1)
```

Comment:
I'd like to avoid optional arguments in public APIs because of potential ambiguities and versioning issues: http://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/